Disable TestF90Callback which fails when compiling the resulting code with ifort.
The resulting code is incorrect Fortran.
See https://github.com/numpy/numpy/issues/20157

Åke Sandgren, 20211021
Updated: Jakob Schiøtz, 20230927
--- numpy-1.25.1/numpy/f2py/tests/test_callback.old	2023-09-27 14:15:21.076625096 +0200
+++ numpy-1.25.1/numpy/f2py/tests/test_callback.py	2023-09-27 14:15:46.401827839 +0200
@@ -202,7 +202,7 @@
     options = ["-DF2PY_USE_PYTHON_TLS"]
 
 
-class TestF90Callback(util.F2PyTest):
+class Disabled_for_intel_TestF90Callback(util.F2PyTest):
     sources = [util.getpath("tests", "src", "callback", "gh17797.f90")]
 
     def test_gh17797(self):
